health/health.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package health
+
+import (
+	"sync"
+	"time"
+
+	"github.com/vechain/thor/v2/thor"
+)
+
+type BlockIngestion struct {
+	BestBlock                   *thor.Bytes32 `json:"bestBlock"`
+	BestBlockIngestionTimestamp *time.Time    `json:"bestBlockIngestionTimestamp"`
+}
+
+type Status struct {
+	Healthy           bool            `json:"healthy"`
+	BlockIngestion    *BlockIngestion `json:"blockIngestion"`
+	ChainBootstrapped bool            `json:"chainBootstrapped"`
+}
+
+type Health struct {
+	lock            sync.RWMutex
+	newBestBlock    time.Time
+	bestBlockID     *thor.Bytes32
+	bootstrapStatus bool
+}
+
+func (h *Health) NewBestBlock(ID thor.Bytes32) {
+	h.lock.Lock()
+	defer h.lock.Unlock()
+
+	h.newBestBlock = time.Now()
+	h.bestBlockID = &ID
+}
+
+func (h *Health) Status() (*Status, error) {
+	h.lock.RLock()
+	defer h.lock.RUnlock()
+
+	blockIngest := &BlockIngestion{
+		BestBlock:                   h.bestBlockID,
+		BestBlockIngestionTimestamp: &h.newBestBlock,
+	}
+
+	healthy := time.Since(h.newBestBlock) <= 10*time.Second && // less than 10 secs have passed since a new block was received
```

Comment:
Will this fail if the node is not in a UTC timezone?

Also, should we give some buffer in case we miss slots?

Another possible solution could be to get all the Block IDs from the peers, and find something like:
- Highest AND
- most common

Also, not sure if this is a concern, but solo can have an interval not equal to 10 seconds. I think the `blockInterval` is passed into some structs

Reply:
Added the healthcheck for solo mode !

Checking the blocks from peers is a more elegant solution, (it's actually how the bootstrap is working) but given that it would require a lot more changes I think it's possible to enhance this feature in the next iteration ?

Reply:
Sure, added a test for the solo 👍